docs/marionette.application.md

```diff
@@ -198,6 +198,10 @@ app.start({
 The Marionette Application provides helper methods for managing its attached
 region.
 
+### `initialize`
```

Comment:
TODO: Need to figure out where this API should be documented as it'll be duplicated in other places, but it isn't common across everything either